src/init/mod.rs

```diff
@@ -162,41 +165,40 @@ fi"#,
             );
             Some(script)
         }
-        Some("ion") => {
+        "ion" => {
             let script = format!("eval $({} init ion --print-full-init)", starship.sprint()?);
             Some(script)
         }
-        Some("elvish") => {
+        "elvish" => {
             let script = format!(
                 "eval (\"{}\" init elvish --print-full-init | slurp)",
                 starship.sprint_posix()?
             );
             Some(script)
         }
-        Some("tcsh") => {
+        "tcsh" => {
             let script = format!(
                 r#"eval "`("{}" init tcsh --print-full-init)`""#,
                 starship.sprint_posix()?
             );
             Some(script)
         }
-        None => {
-            println!(
-                "Invalid shell name provided: {}\\n\
-                 If this issue persists, please open an \
-                 issue in the starship repo: \\n\
-                 https://github.com/starship/starship/issues/new\\n\"",
-                shell_name
-            );
-            None
-        }
-        Some(shell_basename) => {
+        _ => {
+            let quoted_arg = shell_words::quote(shell_basename);
             println!(
-                "printf \"\\n{0} is not yet supported by starship.\\n\
-                 For the time being, we support bash, zsh, fish, and ion.\\n\
+                "printf \"\\n%s is not yet supported by starship.\\n\
+                 For the time being, we support the following shells:\\n\
+                 * bash\\n\
+                 * elvish\\n\
+                 * fish\\n\
+                 * ion\\n\
+                 * powershell\\n\
+                 * tcsh\\n\
+                 * zsh\\n\
+                 \\n\
                  Please open an issue in the starship repo if you would like to \
-                 see support for {0}:\\nhttps://github.com/starship/starship/issues/new\"\\n\\n",
-                shell_basename
+                 see support for %s:\\nhttps://github.com/starship/starship/issues/new\\n\\n\" {0} {0}",
+                quoted_arg
             );
```

Comment:
There is this wonderful macro crate https://github.com/dtolnay/indoc we could leverage to make it a bit more readable IMO :tada: 
```rust
printdoc! {
    r#"printf "\n%s is not yet supported by starship.
    For the time being, we support the following shells:
    * bash
    * elvish
    * fish
    * ion
    * powershell
    * tcsh
    * zsh
    
    Please open an issue in the starship repo if you would like to
    see support for %s:
    https://github.com/starship/starship/issues/new\n\n" {0} {0}"#,
    quoted_arg
};
```

Reply:
I think this would still need literal `\n` everywhere.

Reply:
Nope, it is just one large string spanning multiple lines. You can do this in shell, just tried.

Reply:
Here is the code: https://gist.github.com/vladimyr/16701e99f95c3ec1c0b7b489a2e9d3eb

And the output:
```
$ cargo run 2>/dev/null | sh 

wsh is not yet supported by starship.
For the time being, we support the following shells:
* bash
* elvish
* fish
* ion
* powershell
* tcsh
* zsh

Please open an issue in the starship repo if you would like to
see support for wsh:
https://github.com/starship/starship/issues/new


```

Reply:
I obviously put an extra line break after _would like to_ :facepalm:

Reply:
Well this does actually seem to break it for at least tcsh and elvish.

Reply:
Hmm, right, just got an unmatched quote error in `tcsh`. :disappointed: 

With that being said this shouldn't ever get print inside tcsh or elvsih but more importantly how do we even know that undetected shell supports `printf` in this form? Should we try to invoke `sh` here? :thinking:

Reply:
I think `printf` is reasonably common both as a builtin and as a binary in `$PATH` so there shouldn't be much need for that. This would also require escaping the whole command again.

Reply:
Ah, yeah, forgot about POSIX `print` binary 👍
What can I say - I tried 🙃